### PR TITLE
feat: add support for experimental downloads

### PR DESCRIPTION
**Pull request by MiniDigger**

### CODE REVIEW

File: src/lib/context/downloads.ts [hunk 1]
```diff
@@ -38,21 +41,41 @@ const isVersionStable = async (
...
       }
+      if (
+        !latestExperimentalVersion &&
+        (await isVersionExperimentalStable(id, versions[i]))
+      ) {
+        latestExperimentalVersion = versions[i];
+      }
     }
 
     const project = {
```
Comment (Cubxity) [inline src/lib/context/downloads.ts:75]: Can we not assign `latestExperimentalVersion` when `latestStableVersion != versions[versions.length - 1]`? That way we can avoid making more network requests

File: src/components/layout/SoftwareDownload.tsx [hunk 2]
```diff
@@ -40,11 +54,28 @@ const SoftwareDownload = ({
...
+            <span className="text-blue-600">{version}</span>
           </h2>
           <p className="text-xl mt-4">{description}</p>
-          <div className="flex flex-row gap-4 mt-8">
+          <div className="flex flex-col gap-4 mt-8">
             <SoftwareDownloadButton />
+            {(() => {
+              if (stable && project.latestExperimentalVersion) {
+                return (
+                  <div className="rounded-lg flex flex-row w-full md:w-100 bg-red-500 transition-shadow text-white transition-color hover:shadow-lg pl-5 py-3 cursor-pointer">
```
Comment (Cubxity) [inline src/components/layout/SoftwareDownload.tsx:65]: These elements should be of `button` type

**Review by Cubxity — CHANGES_REQUESTED** [review]

File: src/components/data/SoftwareBuilds.tsx [hunk 3]
```diff
@@ -38,7 +38,10 @@ const SoftwareBuilds = ({
               )}
               rel="noreferrer"
               target="_blank"
-              className="bg-gray-800 text-gray-100 text-sm text-center font-medium rounded-full p-2 min-w-16 mr-4"
+              className={
+                "text-gray-100 text-sm text-center font-medium rounded-full p-2 min-w-16 mr-4 " +
+                (build.channel === "default" ? "bg-gray-800" : "bg-red-500")
+              }
```
Comment (MiniDigger) [inline src/components/data/SoftwareBuilds.tsx:44]: this doesn't feel like the best way to do stuff, is there a more react way for conditional classes like this?

File: src/components/layout/SoftwareDownload.tsx [hunk 4]
```diff
@@ -12,22 +13,44 @@ export interface SoftwareDownloadProps {
...
-  const { data: builds } = useVersionBuilds(id, project.latestVersion);
+  const { data: stableBuilds } = useVersionBuilds(
+    id,
+    project.latestStableVersion
+  );
+  // todo how do I avoid this call when latestExperimentalVersion is null?
+  const { data: experimentalBuilds } = useVersionBuilds(
+    id,
+    project.latestExperimentalVersion
+  );
```
Comment (MiniDigger) [inline src/components/layout/SoftwareDownload.tsx:34]: > ESLint: React Hook "useVersionBuilds" is called conditionally. React Hooks must be called in the exact same order in every component

idk what that is about

File: src/components/data/SoftwareBuilds.tsx (hunk 3 above)
Comment (Cubxity) [reply src/components/data/SoftwareBuilds.tsx:44]: You can use `clsx("class names", condition ? "a" : "b")`

File: src/components/input/SoftwareDownloadButton.tsx [hunk 5]
```diff
@@ -8,21 +8,27 @@ import { DownloadsContext } from "@/lib/context/downloads";
...
+    useContext(DownloadsContext);
 
   const latestBuild = builds && builds[builds.length - 1];
 
   return (
     <Menu as="div" className="relative w-full">
-      <div className="rounded-lg flex flex-row w-full md:w-100 bg-blue-600 transition-shadow text-white transition-color hover:shadow-lg">
+      <div
+        className={
+          "rounded-lg flex flex-row w-full md:w-100 transition-shadow text-white transition-color hover:shadow-lg " +
```
Comment (Cubxity) [inline src/components/input/SoftwareDownloadButton.tsx:20]: Use `clsx`

**Review by Cubxity — CHANGES_REQUESTED** [review]

**Pull request merged**